File: js/24.js
/* 
* @Author: ben_cripps
* @Date:   2014-12-21 18:26:16
* @Last Modified by:   ben_cripps
* @Last Modified time: 2014-12-21 18:40:33
*/

'use strict';

function getPermutations(n) {
  var set = [];

  function permute (arr, data) {
    var cur, memo = data || [];

    for (var i = 0; i < arr.length; i++) {
       cur = arr.splice(i, 1)[0];
       if (arr.length === 0) set.push(memo.concat([cur]));
       permute(arr.slice(), memo.concat([cur]));
       arr.splice(i, 0, cur);
    }
    return set;
 }

 return permute(n);
}

console.log(getPermutations([0,1,2,3,4,5,6,7,8,9])[1000000].join(''));

File: python/24.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2015-01-07 22:26:27
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-07 22:35:34

from itertools import permutations

def get_permutations(str):
    return [x for x in permutations(str, 9)]

def clean(n):
    return ''.join(list(n))

print(clean(get_permutations("0123456789")[1000000]))